libs/src/sdk/services/EntityManager/EntityManager.ts

```diff
@@ -107,8 +101,63 @@ export class EntityManager implements EntityManagerService {
 
     const jsonResponse = await response.json()
 
+    if (!skipConfirmation) {
+      await this.confirmWrite({
+        blockHash: jsonResponse.receipt.blockHash,
+        blockNumber: jsonResponse.receipt.blockNumber,
+        confirmationTimeout
+      })
+    }
+
     return {
       txReceipt: jsonResponse.receipt
     }
   }
+
+  /**
+   * Confirms a write by polling for the block to be indexed by the selected
+   * discovery node
+   */
+  async confirmWrite({
+    blockHash,
+    blockNumber,
+    confirmationTimeout = CONFIRMATION_TIMEOUT
+  }: {
+    blockHash: string
+    blockNumber: number
+    confirmationTimeout?: number
+  }) {
+    const confirmBlock = async () => {
+      const endpoint =
+        await this.config.discoveryNodeSelector.getSelectedEndpoint()
+      const {
+        data: { block_passed }
+      } = await (
+        await fetch(
```

Comment:
Just double checking - we're ok having this bypass our reselection logic?

Reply:
Ehhh actually probably not, let me update

Reply:
Actually gonna think on this one a bit more because it's kinda tricky to call `request` or use middleware in this service

Reply:
Agreed...